command/tls_cert_create.go

```diff
@@ -0,0 +1,298 @@
+package command
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/nomad/helper/flags"
+	"github.com/hashicorp/nomad/helper/tlsutil"
+	"github.com/hashicorp/nomad/lib/file"
+	"github.com/posener/complete"
+)
+
+type TLSCertCreateCommand struct {
+	Meta
+
+	// dnsNames is a list of additional dns records to add to the SAN addresses
+	dnsNames flags.StringFlag
+
+	// ipAddresses is a list of additional IP address records to add to the SAN
+	// addresses
+	ipAddresses flags.StringFlag
+
+	// ca is used to set a custom CA certificate to create certificates from.
+	ca string
+
+	cli    bool
+	client bool
+
+	// key is used to set the custom CA certificate key when creating certificates.
+	key string
+
+	// days is the number of days the certificate will be valid for.
+	days int
+
+	// cluster_region is used to add the region name to the certifacte SAN records
+	cluster_region string
+
+	// domain is used to provide a custom domain for the certificate.
+	domain string
+
+	server bool
+}
+
+func (c *TLSCertCreateCommand) Help() string {
+	helpText := `
+Usage: nomad tls cert create [options]
+
+Create a new certificate
+
+$ nomad tls cert create -server
+==> WARNING: Server Certificates grants authority to become a
+    server and access all state in the cluster including root keys
+    and all ACL tokens. Do not distribute them to production hosts
+    that are not server nodes. Store them as securely as CA keys.
+==> Using nomad-agent-ca.pem and nomad-agent-ca-key.pem
+==> Server Certificate saved to: global-server-nomad.pem
+==> Server Certificate key saved to: global-server-nomad-key.pem
+$ nomad tls cert create -client
+==> Using CA file nomad-agent-ca.pem and CA key nomad-agent-ca-key.pem
+==> Client Certificate saved to global-client-nomad.pem
+==> Client Certificate key saved to global-client-nomad-key.pem
```

Comment:
Having examples in the docstring is going to look confusing, especially with the big WARNING flags.